jobsdb/jobsdb.go

```diff
@@ -2390,26 +2409,26 @@ func (jd *HandleT) doStoreJobsInTx(ctx context.Context, tx *sql.Tx, ds dataSetT,
 
 func (jd *HandleT) storeJob(ctx context.Context, tx *sql.Tx, ds dataSetT, job *JobT) (err error) {
 	sqlStatement := fmt.Sprintf(`INSERT INTO %q (uuid, user_id, custom_val, parameters, event_payload, workspace_id)
-	                                   VALUES ($1, $2, $3, $4, $5, $6) RETURNING job_id`, ds.JobTable)
+		VALUES ($1, $2, $3, $4, $5, $6) RETURNING job_id`, ds.JobTable)
 	stmt, err := tx.PrepareContext(ctx, sqlStatement)
-	jd.assertError(err)
-	defer func() { _ = stmt.Close() }()
+	if err != nil {
+		return err
+	}
+	defer stmt.Close()
 	job.sanitizeJson()
 	_, err = stmt.ExecContext(ctx, job.UUID, job.UserID, job.CustomVal, string(job.Parameters), string(job.EventPayload), job.WorkspaceId)
-	if err == nil {
-		// Empty customValFilters means we want to clear for all
-		jd.markClearEmptyResult(ds, allWorkspaces, []string{}, []string{}, nil, hasJobs, nil)
-		jd.markClearEmptyResult(ds, job.WorkspaceId, []string{}, []string{}, nil, hasJobs, nil)
-		// fmt.Println("Bursting CACHE")
-		return
-	}
-	pqErr, ok := err.(*pq.Error)
-	if ok {
-		errCode := string(pqErr.Code)
+
+	var e *pq.Error
+	if err != nil && errors.As(err, &e) {
+		errCode := string(e.Code)
 		if _, ok := dbInvalidJsonErrors[errCode]; ok {
 			return errors.New("invalid JSON")
 		}
 	}
+
+	// Empty customValFilters means we want to clear for all
+	jd.markClearEmptyResult(ds, allWorkspaces, []string{}, []string{}, nil, hasJobs, nil)
+	jd.markClearEmptyResult(ds, job.WorkspaceId, []string{}, []string{}, nil, hasJobs, nil)
```

Comment:
[question] The order of the statements has been reversed, hope this is intended

Reply:
reverted back.